File: Sorting/SelectionSort/selectionsort.cpp
#include <iostream>
using namespace std;



void swap(int *a,int *b){
    int temp = *a;
    *a = *b;
    *b = temp;
}

void selectionSort(int arr[],int n){
 int min;
 int temp;
 for ( int i =0 ; i<n-1;i++){
    min = i;
    for(int j=i+1;j<n;j++){
        if (arr[min]>arr[j]){
            min = j;
        }
    }
    if(min!=i){
        temp = arr[i];
        arr[i] = arr[min];
        arr[min] = temp;
        // swap(&arr[min],&arr[i]);
    }

 }
}
int main(){
    int arr[] ={2,3,4,5,0,6,7,8,9};
    int n = sizeof(arr)/sizeof(arr[0]);
    selectionSort(arr,n);
    for ( int i=0 ; i<n;i++){
        cout<<arr[i]<<endl;
    }
    return 0;
}

File: Recursion/selectionsort.py

def findMin(a,l,h):
    low  = l
    min = 9999999999
    ans_index = -1
    while low < h :
        if a[low] < min :

            min = a[low]
            ans_index = l
        low+=1
    a[l],a[ans_index] = a[ans_index],a[l]



def  selectionsort(a,l,h):
    if l == h :
        return 
    else:
        findMin(a,l,h)
        selectionsort(a,l+1,h)

arr =[20,1,2,4,30]
selectionsort(arr,0,len(arr))